fretboard-client-kinto/src/main/java/mozilla/components/service/fretboard/source/kinto/HttpClient.kt

```diff
@@ -4,14 +4,15 @@
 
 package mozilla.components.service.fretboard.source.kinto
 
+import java.io.Serializable
 import java.net.URL
 
 /**
  * Represents an http client, used to
  * make it easy to swap implementations
  * as needed
  */
-interface HttpClient {
+interface HttpClient : Serializable {
```

Comment:
Oh, are we marking all those classes as Serializable because we need to pass them to the Job(Service)? This can be error-prone and we may get into problems once one of those classes is not ``Serializable``. Also you may end up with duplicated/copied objects on the other side - even though you are still in the same process.

Let's try something else instead: Can we make `SyncJob` abstract and add an abstract method that provides an `Fretboard` instance to it? This would allow a consumer to extend the service class, add it to the manifest and implement `getFretboard()`. Different apps may have different ways of how to keep track of the Fretboard instance and this would allow them to keep whatever technique they use (e.g. dependency injection).